lib/chrono_model/patches/relation.rb

```diff
@@ -37,7 +37,11 @@ def chrono_join_history(join)
         #
         return if join.left.respond_to?(:as_of_time)
 
-        model = ChronoModel.history_models[join.left.table_name]
+        if (join.left.respond_to?(:table_name))
+          model = ChronoModel.history_models[join.left.table_name]
+        else
+          model = ChronoModel.history_models[join.left]
+        end
```

Comment:
I propose this cosmetic change

```rb
model =
  if join.left.respond_to?(:table_name)
    ChronoModel.history_models[join.left.table_name]
  else
    ChronoModel.history_models[join.left]
  end
```

Reply:
i totally agree

Reply:
![image](https://user-images.githubusercontent.com/223344/140771415-a43eaf67-c34b-4b96-bb09-fd34ce19cbe0.png)